### PR TITLE
CB-13673 support of preferred availability zones for datahub scaling.

### PR DESCRIPTION
**Pull request by biharitomi**
At scaling the user could decide which availability zones should be preferred by control plane at instance provisioning.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/request/InstanceGroupAdjustmentV4Request.java [hunk 1]
```diff
@@ -20,6 +22,9 @@
     @ApiModelProperty(value = InstanceGroupAdjustmentModelDescription.SCALING_ADJUSTMENT, required = true)
     private Integer scalingAdjustment;
 
+    @ApiModelProperty(value = ModelDescriptions.InstanceGroupNetworkScaleModelDescription.NETWORK_SCALE_REQUEST)
+    private StackNetworkScaleV4Request stackNetworkScaleV4Request;
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/request/InstanceGroupAdjustmentV4Request.java:26]: how about simply networkScaleV4Request

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/StackScaleV4Request.java [hunk 2]
```diff
@@ -28,6 +30,9 @@
     @ApiModelProperty(InstanceGroupAdjustmentModelDescription.FORCE)
     private Boolean forced;
 
+    @ApiModelProperty(value = InstanceGroupNetworkScaleModelDescription.NETWORK_SCALE_REQUEST)
+    private StackNetworkScaleV4Request stackNetworkScaleV4Request;
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/StackScaleV4Request.java:34]: how about simply networkScale

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/network/StackNetworkScaleV4Request.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class StackNetworkScaleV4Request implements JsonEntity {
+
+    @ApiModelProperty(value = ModelDescriptions.InstanceGroupNetworkScaleModelDescription.PREFERRED_SUBNET_IDS)
+    private List<String> preferredSubnetIds = new ArrayList<>();
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/network/StackNetworkScaleV4Request.java:18]: subnetIds ?

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java [hunk 4]
```diff
@@ -22,6 +24,9 @@
     @ApiModelProperty(value = InstanceGroupAdjustmentModelDescription.SCALING_ADJUSTMENT, required = true)
     private Integer desiredCount;
 
+    @ApiModelProperty(value = InstanceGroupNetworkScaleModelDescription.NETWORK_SCALE_REQUEST)
+    private DistroXNetworkScaleV1Request networkScaleV1Request;
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/DistroXScaleV1Request.java:28]: how about simply networkScale

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/network/DistroXNetworkScaleV1Request.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DistroXNetworkScaleV1Request implements JsonEntity {
+
+    @ApiModelProperty(value = ModelDescriptions.InstanceGroupNetworkScaleModelDescription.PREFERRED_SUBNET_IDS)
+    private List<String> preferredSubnetIds = new ArrayList<>();
```
Comment (doktoric) [inline core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/network/DistroXNetworkScaleV1Request.java:18]: subnetIds ?

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/dto/StackNetworkScaleDetails.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package com.sequenceiq.cloudbreak.core.flow2.dto;
+
+import java.util.Collections;
+import java.util.List;
+
+public class StackNetworkScaleDetails {
+
+    private final List<String> preferredSubnetIds;
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/dto/StackNetworkScaleDetails.java:8]: subnetIds

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/network/StackNetworkScaleV4Request.java (hunk 3 above)
Comment (biharitomi) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/network/StackNetworkScaleV4Request.java:18]: I am fine with preferredSubnetIds.

File: core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/network/DistroXNetworkScaleV1Request.java (hunk 5 above)
Comment (biharitomi) [reply core-api/src/main/java/com/sequenceiq/distrox/api/v1/distrox/model/network/DistroXNetworkScaleV1Request.java:18]: I am fine with preferredSubnetIds.

**Review by doktoric — APPROVED** [review]

**Pull request merged**